### ISSUE TITLE
AttributeError: 'Distribution' object has no attribute 'executables' during cx_freeze process with 6.14 version

### ISSUE THREAD

**Issue by sikandarilyas** [marcelotduarte/cx_Freeze#1848]
**Describe the bug**
We are using cx_freeze for freezing our python application. Recently we planned to move to Python 3.11 and hence picked up CX_FREEZE version 6.14 along with it. While freezing our application with 6.14 version we saw a failure as below:-

error: Multiple top-level packages discovered in a flat-layout: ['build_1', 'build_2']

I found documentation that this is because of auto-discovery mechanism and I used packages=[] to switch if off. Once done that I got new errors as below.

AttributeError: 'Distribution' object has no attribute 'executables'.

Looking more into it I found that the Distribution mechanism is updated in cx_freeze and now setuptools Distribution is used. I tried with older version 6.13 of cx_freeze and it worked without any such errors.

I found below commits around the change in Distribution mechanism:-
https://github.com/marcelotduarte/cx_Freeze/commit/7b43bb8c5a495ce353179a1aa2728ed7481f165e
https://github.com/marcelotduarte/cx_Freeze/commit/61be7474fc136c0d78a5f545078c38041e73ee9c
https://github.com/marcelotduarte/cx_Freeze/commit/8f8b19ba59c1c16ccfc8e0559b47e5db9b152aca

Also I found below change which updates string_list_options to list_options andI was getting another error because of it.
https://github.com/marcelotduarte/cx_Freeze/commit/75da6a45c9f0c0c79abd7b2ee4336ead5db2d44d

I reverted above changes in cx_freeze build and after that 6.14 version started working fine. I could generate a EXE with these changes. Now I am not sure if this is a real problem or we need to adjust out setup.py file with some new attributes but I couldn't find any new changes as per the documentation of 6.14 version.

**To Reproduce**
I am using a sample setup.py as added in the documentation page. No more code in setup.py, just copied the code as-is from the documentation of cx_freeze.

**Expected behavior**
I should be able to generate exe with cx_freeze 6.14 version without any changes from our side.

**Screenshots**
![image](https://user-images.githubusercontent.com/95215225/227172846-2fd57a2d-e43f-4c3a-9901-521e091ed76a.png)

**Desktop (please complete the following information):**
 - Windows 10
 - cx_Freeze version 6.14
 - Python version 3.11

**Comment by marcelotduarte** [comment]
V 6.14 has a bug and was yanked, use 6.14.7 instead. Use the setuptools 65.x
Dont use packages=[], cx_Freeze uses py_modules = [] to bypass auto-discovery.

**Comment by sikandarilyas** [comment]
Thanks @marcelotduarte for update. I do not see any official confirmation about not using 6.14. Release notes still point to 6.14 as the latest release. Also cx_version compatible with Python 3.11 is updated as 6.14. I am looking at below links:-

https://cx-freeze.readthedocs.io/en/6.14.1/releasenotes.html
https://cx-freeze.readthedocs.io/en/6.14.1/versions.html

Is there a plan to update these links? Is there any other place (forum etc.) where we should look to get latest information like the one you shared here.

I will use cx_freeze 6.14.7 and update here later.

**Comment by marcelotduarte** [comment]
Only on PyPI has info: https://pypi.org/project/cx-Freeze/#history

**Comment by sikandarilyas** [comment]
Hi @marcelotduarte , I tried using cx_freeze 6.14.7 with setuptools 65.5 but getting same attribute error. This time I updated setup.py with py_modules=[]. 
As posted initially, unless I revert above commits, cx_freeze doesn't work. What may be wrong here? Please suggest.

**Comment by marcelotduarte** [comment]
Show me your setup.

**Comment by sikandarilyas** [comment]
Hi @marcelotduarte . Thanks for looking into it. Below is my setup.py (almost similar to the example as in cx_freeze documentation).

**setup.py**:-
```
import os
import sys
#os.environ["_CX_FREEZE"] = "C:\\Users\\ilyass\\Desktop\\win64"
os.environ["_CX_FREEZE"] = "C:\\Git_Depot\\Streamer\\3P\\CX_FREEZE\\6.14\\WIN64_1"
#os.environ["_CX_FREEZE"] = "C:\\Users\\ilyass\\\Desktop\cxfreeze_6.4"
sys.path.append(os.path.join(os.environ['_CX_FREEZE'], 'lib'))

from cx_Freeze import setup, Executable

#Dependencies are automatically detected, but it might need fine tuning.
#"packages": ["os"] is used as example only
build_exe_options = {"packages": ["os"], "excludes": ["tkinter"]}

#base="Win32GUI" should be used only for Windows GUI app
base = None
if sys.platform == "win32":
    base = "Win32GUI"

setup(
    name="guifoo",
    description="My GUI application!",
    options={"build_exe": build_exe_options},
    py_modules = [],
    executables=[Executable("guifoo.py", base=base)],
)
```
We are calling setup.py from a helper script StartCXFreeze.py. Here is the code for it.

**StartCXFreeze.py**:-
```
import sys, platform
if platform.system() == "Windows":
    #buildType = ""
    buildType = "bdist_msi"
else:
    buildType = "bdist_mac"
sys.argv.append(buildType)
import setup
sys.argv.remove(buildType)
```
We are using python 3.11.1. Let know in case any more information is required.

**Comment by marcelotduarte** [comment]
You can remove py_modules, cx_Freeze already add it.
If you run setup.py directly, it works?
But I belied that the error is in line 6 sys.path.append, that changes the behavior of search. Can you remove it to test?

**Comment by sikandarilyas** [comment]
Hi @marcelotduarte , removing the line sys.path.append works. There is no error now.

Question is , how the same setup worked for version 6.13 and not for 6.14? We need to add sys.path.append to search for cx_Freeze module since we do not add it to the "site_packages" in python "Libs".

Is there a way we can get 6.14.7 to work without removing that line? 

FYI....I also removed py_modules=[].

**Comment by marcelotduarte** [comment]
I make some changes to the full support configuration using pyproject.toml. It is partial support yet.
As cx_Freeze inherits classes from setuptools, cx_Freeze needs to be installed.
How you have used cx_Freeze before, how did you install it? Using pip install --platform ....?

**Comment by sikandarilyas** [comment]
Hi @marcelotduarte , Previously, we have used cx_Freeze from another path and added the path to sys.path so that it can be searched. We did not install cx_freeze with pip. 

This approach worked fine with 6.13 but seems broken with 6.14.7. Are you suggesting we need to now install cx_Freeze with pip to make it work? Is this the only way to get it working in 6.14.7 as sys.path.append seems to error out.
